File: Code/Codechef/start183/E.py
N = 3
M = 2
def solve():
    results = []
    for N, M in test_cases:
        K = N - 1
        T = N * (N - 1) // 2
        Z = T - M
        min_W = max(0, K - Z)
        max_W = min(K, M)
        if min_W > max_W:
            results.append(0)
        else:
            count = max_W - min_W + 1
            total = (min_W + max_W) * count // 2
            results.append(total)
    return results

# Reading input
T = int(input())
test_cases = [tuple(map(int, input().split())) for _ in range(T)]
for res in solve(test_cases):
    print(res)

File: Code/Codechef/start183/E.cpp
#include <iostream>
using namespace std;

typedef long long ll;

ll computeSumW(ll N, ll M) {
    ll totalEdges = N * (N - 1) / 2;
    ll zeroEdges = totalEdges - M;
    ll need = N - 1;

    ll minW = max(0LL, need - zeroEdges);
    ll maxW = min(need, M);

    if (minW > maxW) return 0;

    ll count = maxW - minW + 1;
    ll sum = (minW + maxW) * count / 2;
    return sum;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;
    while (T--) {
        ll N, M;
        cin >> N >> M;
        cout << computeSumW(N, M) << "\n";
    }

    return 0;
}
